src/libs/metadata_view.c

```diff
@@ -791,6 +807,115 @@ void connect_key_accels(dt_lib_module_t *self)
+  for(int i = 0; i < md_size; i++)
+  {
+    const gboolean hidden = g_list_find_custom(prefs, _labels[i], (GCompareFunc)g_strcmp0) != NULL;
+    for(int j = 0; j < 2; j++)
+    {
+      GtkWidget *w = gtk_grid_get_child_at(GTK_GRID(d->grid), j, i);
+      gtk_widget_show_all(w);
+      gtk_widget_set_no_show_all(w, TRUE);
+      gtk_widget_set_visible(w, !hidden);
+    }
+  }
+  g_list_free_full(prefs, g_free);
+}
+
+void set_preferences(dt_lib_module_t *self)
+{
+  dt_lib_metadata_view_t *d = (dt_lib_metadata_view_t *)self->data;
+
+  GtkWidget *win = dt_ui_main_window(darktable.gui->ui);
+  GtkWidget *dialog = gtk_dialog_new_with_buttons(_("metadata settings"), GTK_WINDOW(win), GTK_DIALOG_DESTROY_WITH_PARENT,
+                                       _("cancel"), GTK_RESPONSE_NONE, _("save"), GTK_RESPONSE_YES, NULL);
+  GtkWidget *area = gtk_dialog_get_content_area(GTK_DIALOG(dialog));
+
+  GtkWidget *w = gtk_scrolled_window_new(NULL, NULL);
+  gtk_widget_set_size_request(w, -1, DT_PIXEL_APPLY_DPI(300));
+  gtk_scrolled_window_set_policy(GTK_SCROLLED_WINDOW(w), GTK_POLICY_AUTOMATIC, GTK_POLICY_AUTOMATIC);
+  gtk_container_add(GTK_CONTAINER(area), w);
+
+  GtkWidget *grid = gtk_grid_new();
+  gtk_container_add(GTK_CONTAINER(w), grid);
+  GtkWidget *label = gtk_label_new(_("metadata"));
+  gtk_grid_attach(GTK_GRID(grid), label, 0, 0, 1, 1);
+  label = gtk_label_new(_("hidden"));
+  gtk_grid_attach(GTK_GRID(grid), label, 1, 0, 1, 1);
+  gtk_widget_set_tooltip_text(GTK_WIDGET(label),
+            _("tick if the corresponding metadata is of no interest for you"));
+
+  char *pref = dt_conf_get_string("plugins/lighttable/metadata_view/pref_hidden");
+  GList *prefs = dt_util_str_to_glist(",", pref);
+  g_free(pref);
+  GtkWidget *hidden[md_size];
+  gboolean hide[md_size];
+  for(int i = 0; i < md_size; i++)
+  {
+    label = gtk_label_new(_md_labels[i]);
+    gtk_widget_set_halign(label, GTK_ALIGN_START);
+    gtk_label_set_xalign (GTK_LABEL(label), 0.0f);
+    gtk_label_set_ellipsize(GTK_LABEL(label), PANGO_ELLIPSIZE_END);
+    gtk_grid_attach(GTK_GRID(grid), label, 0, i+1, 1, 1);
+
+    hidden[i] = gtk_check_button_new();
+    gtk_widget_set_halign(hidden[i], GTK_ALIGN_CENTER);
+    hide[i] = g_list_find_custom(prefs, _labels[i], (GCompareFunc)g_strcmp0) != NULL;
+    gtk_toggle_button_set_active(GTK_TOGGLE_BUTTON(hidden[i]), hide[i]);
+    gtk_grid_attach(GTK_GRID(grid), hidden[i], 1, i+1, 1, 1);
+    if(i >= md_xmp_metadata && i < md_xmp_metadata + DT_METADATA_NUMBER)
+    {
+      // metadata plugin preferences have precedence
+      if(!gtk_widget_is_visible(GTK_WIDGET(d->name[i])))
+      {
+        gtk_widget_show_all(label);
+        gtk_widget_set_no_show_all(label, TRUE);
+        gtk_widget_set_visible(label, FALSE);
+        gtk_widget_show_all(hidden[i]);
+        gtk_widget_set_no_show_all(hidden[i], TRUE);
+        gtk_widget_set_visible(hidden[i], FALSE);
+      }
+    }
+  }
+  gtk_window_set_resizable(GTK_WINDOW(dialog), FALSE);
```

Comment:
With the gtk_box_pack_start and gtk_widget_set_hexpand changes above, the dialog can be safely resized. Which is useful to be able to read the title and longer field names without ellipsizing.

Reply:
I wanted to do that, but haven't take the time to find out how. I'll do it.

Reply:
Works indeed !